dev-python/QtPy/QtPy-1.9.0-r4.ebuild

```
@@ -0,0 +1,117 @@
+#
+# Webengine is a special case, because PyQt5 provides this in a separate package
+# while PySide2 ships it in the same package.
+#
+# declarative/qml/quick is a special case, because PyQt5 bundles the bindings
+# for qml and quick in one flag: declarative PySide2 does not.
+#
+# The PyQt5 ebuild currently enables xml support unconditionally, the flag is
+# added anyway with a (+) to make it future proof if the ebuild were to change
+# this behaviour in the future.
+#
+# The PySide2 ebuild currently enables opengl and serialport support
+# unconditionally, the flag is added anyway with a (+) to make it future proof
+# if the ebuild were to change this behaviour in the future.
+#
+RDEPEND="
+	pyqt5? (
+		dev-python/PyQt5[${PYTHON_USEDEP},bluetooth?,dbus?,designer?,declarative?,gui?,help?,location?,multimedia?,network?,networkauth?,opengl?,positioning?,printsupport?,sensors?,serialport?,sql?,ssl?,svg?,testlib?,webchannel?,webkit?,websockets?,widgets?,x11extras?,xml(+)?,xmlpatterns?]
+		webengine? ( dev-python/PyQtWebEngine[${PYTHON_USEDEP}] )
+	)
+	pyside2? (
+		dev-python/pyside2[${PYTHON_USEDEP},3d?,charts?,datavis?,designer?,gui?,help?,location?,multimedia?,network?,opengl(+)?,positioning?,printsupport?,qml?,quick?,script?,scripttools?,scxml?,sensors?,serialport(+)?,speech?,sql?,svg?,testlib?,webchannel?,webengine?,websockets?,widgets?,x11extras?,xml?,xmlpatterns?]
+	)
+"
+
+# The QtPy testsuite skips tests for bindings that are not installed, so here we
+# ensure that everything is available and all tests are run. Note that not
+# all flags are available in PyQt5/PySide2, so some tests are still skipped.
+BDEPEND="
+	test? (
+		dev-python/mock[${PYTHON_USEDEP}]
+		dev-python/PyQt5[${PYTHON_USEDEP},bluetooth,dbus,declarative,designer,gui,help,location,multimedia,network,networkauth,opengl,positioning,printsupport,sensors,serialport,sql,ssl,svg,testlib,webchannel,webkit,websockets,widgets,x11extras,xml(+),xmlpatterns]
+		dev-python/PyQtWebEngine[${PYTHON_USEDEP}]
+		dev-python/pyside2[${PYTHON_USEDEP},3d,charts,concurrent,datavis,designer,gui,help,location,multimedia,network,opengl(+),positioning,printsupport,qml,quick,script,scripttools,scxml,sensors,serialport(+),speech,sql,svg,testlib,webchannel,webengine,websockets,widgets,x11extras,xml,xmlpatterns]
+)"
+
+distutils_enable_tests pytest
+
+src_prepare() {
+	default
+	use pyqt5 || sed -i -e "s/from PyQt5.QtCore import/raise ImportError #/" qtpy/__init__.py || die
+	use pyside || sed -i -e "s/from PySide2 import/raise ImportError #/" qtpy/__init__.py || die
+
+	# Disable outdated PyQt4 and PySide
+	sed -i -e "s/from PyQt4.Qt import/raise ImportError #/" qtpy/__init__.py || die
+	sed -i -e "s/from PyQt4.QtCore import/raise ImportError #/" qtpy/__init__.py || die
+	sed -i -e "s/from PySide import/raise ImportError #/" qtpy/__init__.py || die
+}
+
+python_test() {
+	export QT_API="pyqt5"
+	virtx pytest -vv
+	export QT_API="pyside2"
+	virtx pytest -vv
+	unset QT_API
```

```suggestion
	local -x QT_API
	for QT_API in pyqt5 pyside2; do
		virtx epytest
	done
```

Also, it may be possible to move `virtx` into `src_test` to wrap the whole test phase in a single virtx.